### DISCUSSION TITLE
Adding Parent with Child variations

### DISCUSSION THREAD

**Discussion by 200317314** [abuzuhri/Amazon-SP-API-CSharp#726] [Q&A]
I am adding some HOME product types, first I submit the parent object and store the SKU parent object has these 3 values

` if (!attrs.ContainsKey("parentage_level")) {
                    attrs.Add("parentage_level", new object[] {
                        new {
                            language_tag = "en_CA",
                            value = "parent",
                            marketplace_id = FikaAmazonAPI.Utils.MarketPlace.Canada.ID
                        }
                    });
                }

                if (!attrs.ContainsKey("child_parent_sku_relationship")) {
                    attrs.Add("child_parent_sku_relationship", new object[] {
                        new {
                            language_tag = "en_CA",
                            parent_sku = "",
                            child_relationship_type = "variation",
                            marketplace_id = FikaAmazonAPI.Utils.MarketPlace.Canada.ID
                        }
                    });
                }

                if (!attrs.ContainsKey("variation_theme")) {
                    var variationName = (string.IsNullOrEmpty(size)) ? "COLOR_NAME" : "COLOR_NAME/SIZE_NAME";

                    attrs.Add("variation_theme", new object[] {
                        new {
                            language_tag = "en_CA",
                            name = variationName,
                            marketplace_id = FikaAmazonAPI.Utils.MarketPlace.Canada.ID
                        }
                    });
                }`

Then I create a child that has the Color or Size etc and submit it with these

`if (!attrs.ContainsKey("parentage_level")) {
                    attrs.Add("parentage_level", new object[] {
                        new {
                            language_tag = "en_CA",
                            value = "child",
                            marketplace_id = FikaAmazonAPI.Utils.MarketPlace.Canada.ID
                        }
                    });
                }

                if (!attrs.ContainsKey("child_parent_sku_relationship")) {
                    attrs.Add("child_parent_sku_relationship", new object[] {
                        new {
                            language_tag = "en_CA",
                            parent_sku = parentSku,
                            child_relationship_type = "variation",
                            marketplace_id = FikaAmazonAPI.Utils.MarketPlace.Canada.ID
                        }
                    });
                }

                if (!attrs.ContainsKey("variation_theme")) {
                    var variationName = (string.IsNullOrEmpty(size)) ? "COLOR_NAME" : "COLOR_NAME/SIZE_NAME";

                    attrs.Add("variation_theme", new object[] {
                        new {
                            language_tag = "en_CA",
                            name = variationName,
                            marketplace_id = FikaAmazonAPI.Utils.MarketPlace.Canada.ID
                        }
                    });
                }`

The issue is now I see my parents in the inventory section like this

![image](https://github.com/abuzuhri/Amazon-SP-API-CSharp/assets/31968510/60a06548-43e7-4f3f-84c0-96f56e2c3bb1)

it shows incomplete and when I click them to see what is incomplete i get "This SKU has an inconsistent product classification with the ASIN. Please submit a feed to update this SKU to match the ASIN product classification.  "

In the complete your drafts I see all the children items like this

![image](https://github.com/abuzuhri/Amazon-SP-API-CSharp/assets/31968510/8fec0958-c3d2-4ba5-acb9-9ae62b882469)

in the Edit it shows

![image](https://github.com/abuzuhri/Amazon-SP-API-CSharp/assets/31968510/070fc689-f927-46ce-bf9f-6ce74088ab87)

I am not sure how to fix this as I have included the color into the json and confirmed the child is getting a color selected

Any help?

@TechyChap 
@abuzuhri 
@ElonMusk
@jeffbezos

**Comment by wbaldini** [comment]
hi,
i've this problem now ...

did you solve this problem?

Thank you so much for replying.
Have a good day.
